realm/realm-library/src/objectServer/java/io/realm/SyncUser.java

```diff
@@ -61,14 +60,30 @@
  */
 public class SyncUser {
 
+    private final ObjectServerUser syncUser;
+    private SyncUser(ObjectServerUser user) {
+        this.syncUser = user;
+    }
+    private ThreadLocal<PermissionManager> permissionManager = new ThreadLocal<PermissionManager>() {
+        @Override
+        public PermissionManager get() {
+            PermissionManager pm = super.get();
+            if (pm == null || pm.isClosed()) {
+                pm = new PermissionManager(SyncUser.this);
+                set(pm);
+                return pm;
+            } else {
```

Comment:
I would like to see the smaller clause first.  ... in fact, I don't see why you don't just factor it out...